client/python/gaia/client.py

```diff
@@ -47,11 +47,22 @@ def pop_path(path):
 
     return '/'.join(parts[1:])
 
-def launch_sisyphus(key):
+def launch_sisyphus(options):
     command = os.path.join("script", "launch-sisyphus.sh")
     if not os.path.exists(command):
         command = "launch-sisyphus.sh"
-    os.system("{} {}".format(command, key))
+
+    worker = options.get('worker', 'sisyphus')
+    metadata = options.get('metadata', {})
+
+    launch_metadata = ''
+    if metadata:
+        metadata_fields = []
+        for key, value in metadata.items():
+            metadata_fields.append('{}={}'.format(key, value))
+        launch_metadata = ','.join(metadata_fields)
+
+    os.system("{} {} {}".format(command, worker, launch_metadata))
```

Comment:
On further reflection, let's replace `launch-sisyphus.sh` and `launch-workers.sh` with a Python script that uses `subprocess32.call()` to call `gcloud`, provides a Python API for the workflow builder and others to call, and provides an argparse CLI for interactive debugging.

That'll avoid bash arrays and quoting hell while making it straightforward to pass in metadata fields.

It can get the app-specific parameters (machine-type, service-account, scopes, image-family...) from **kwargs or from an app-specific file.

Reply:
> The worker-launching code in wcEcoli's runscripts/cloud/util/workflow.py + runscripts/cloud/launch-workers.sh moved beyond gaia/client

Ahaha okay, no worries I could have known that as well. I think launching is a gaia/sisyphus functionality so might have to migrate that back. 

Alternatively, I could see if we can use this compute service feature to make the launch call directly from the JVM code? And not have to muck around with shelling out to scripts etc. Python script could work also, whatever works!